jujugui/static/gui/src/app/components/deployment-flow/machines/machines.js

```diff
@@ -42,27 +42,26 @@ YUI.add('deployment-machines', function() {
       Object.keys(machines).forEach(key => {
         const machine = machines[key];
         let constraintsDetails;
-        let seriesDetails = '';
         const args = machine.command.args[0][0];
         const series = args.series;
         const constraints = args.constraints || {};
         const cpu = constraints.cpu;
         const disk = constraints.disk;
         const mem = constraints.mem;
         const cores = constraints.cores;
+        const parts = [];
         if (cores && cpu && disk && mem) {
           cpu = cpu / 100;
           disk = disk / 1024;
           mem = mem / 1024;
           constraintsDetails = `${cores}x${cpu}GHz, ${mem.toFixed(2)}GB, ` +
             `${disk.toFixed(2)}GB`;
-        } else {
-          constraintsDetails = '(constraints not set)';
         }
         if (series) {
-          seriesDetails = `${series}, `;
+          parts.push(`${series}${constraintsDetails ? ',' : ''}`);
```

Comment:
Instead of doing this could you instead do:

```JavaScript
if (series) {
  parts.push(series);
}
parts.push(constraintsDetails || '(constraints not set)');
const info = parts.join(', ');
```